torch/quantization/fx/utils.py

```diff
@@ -345,47 +346,58 @@ def create_qparam_nodes(quantizer: QuantizerCls, node_name: str, scale: Any, zer
     return (scale_node, zero_point_node)
 
 
-def all_node_args_have_no_tensors(node: Node) -> bool:
+def all_node_args_have_no_tensors(node: Node, modules: Dict[str, torch.nn.Module], cache: Optional[Dict[Node, bool]] = None) -> bool:
```

Comment:
optional: can we make cache a required argument, to reduce complexity?

optional: what broke when trying to implement the cache in a wrapper func, and keep the inner func from knowing about the cache?  I feel like it should be possible, since the cache conceptually is just caching the results of a computation.

Reply:
sure I can make cache required.

for wrapper, I tried that, but since we need to pass around cache to the recursive calls, we ends up with the same code